### DISCUSSION TITLE
How to use usesetting argument

### DISCUSSION THREAD

**Discussion by rey-nord** [simnibs/simnibs#228] [7.Creating head models]
Hello,
I am trying to fix spurious segmentation results obtained with charm. Specifically, I am working with an MRI of an old individual which demonstrates some brain atrophy. The segmentation results for inner layer of the skull and CSF are not perfect and require correction.
When I re-run charm for this subject it produces two errors
RuntimeError: ERROR: --forcerun has to be set to overwrite existing m2m_{subID} folder
ValueError: The qform and sform do not match. Please run charm with the --forceqform (preferred) or --forcesform option
Using --forcerun --forceqform resolves the problem

It appears that --forcerun erases all files in m2m_{subID} folder and instruction from the Fixing the Affine Registration suggesting to make a copy of settings.ini doesn't explain much about usesettings argument e.g. where the new setting.ini file can be stored and dosn't give an example of code

I found one similar discussion about usesetting https://github.com/simnibs/simnibs/discussions/208
There the following line was used leading to error 
charm patientNumber(subID) --initatlas --usesettings settings.ini
but the question remained unanswered

Ive also been searching for an answer in several places
https://simnibs.github.io/simnibs/build/html/tutorial/advanced/fix_affine_registration.html#fix-affine-registration-tutorial
https://simnibs.github.io/simnibs/build/html/documentation/command_line/charm.html#charm-docs
https://simnibs.github.io/simnibs/build/html/tutorial/advanced/fix_headmodel.html#fixheadmodel-tutorial
but I was not able to find examples/explanations related to usesettings except the one already mentioned

I would be thankful for info (with example code line if possible) on how to use new setting file with updated settings and –usesettings argument in charm. I believe this might be useful for SimNIBS users.

Thank you!

Andre

**Comment by oulap** [comment]
You can store the settings file wherever you wish, just not inside the m2m-folder charm will overwrite (because then the settings file would be overwritten too). So let's say you copied the setting file to the path /my/path/settings.ini

You could then pass that to charm by calling:

charm sub_id --initatlas --usesettings /my/path/setting.ini

Discussion #208 relates to some other issue with the settings file, not passing it to charm. However, if you give the settings file to charm (with the correct path) and still encounter errors, then your problem might be related to #208 .

**Comment by rey-nord** [comment]
Thank you for the prompt reply!
I was able to resolve the problem and run charm successfully with updated settings.
